cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java

```diff
@@ -100,6 +114,42 @@
     return Collections.emptySet();
   }
 
+  /**
+   * Retrieve topic minISR config information if it is not cached locally.
+   * @param topicsToCheck Set of topics to check.
+   */
+  private void maybeRetrieveAndCacheTopicMinISR(Set<String> topicsToCheck) {
+    List<ConfigResource> topicResourcesToCheck = new ArrayList<>(topicsToCheck.size());
+    topicsToCheck.stream().filter(t -> !_cachedTopicMinISR.containsKey(t))
+                          .forEach(t -> topicResourcesToCheck.add(new ConfigResource(ConfigResource.Type.TOPIC, t)));
+    for (Map.Entry<ConfigResource, KafkaFuture<Config>> entry : _adminClient.describeConfigs(topicResourcesToCheck).values().entrySet()) {
+      try {
+        short topicMinISR = Short.parseShort(entry.getValue().get(METADATA_FETCH_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+                                                  .get(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG).value());
+        _cachedTopicMinISR.put(entry.getKey().name(), new TopicMinISREntry(topicMinISR, System.currentTimeMillis()));
+      } catch (TimeoutException | InterruptedException | ExecutionException e) {
+        LOG.warn("Skip attempt to fix topic {}'s replication factor due to unable to retrieve its minISR config.",
```

Comment:
Nit (just personal style note -- please feel free to ignore): I usually prefer the format `X of Y` rather than `Y's X`, which makes it easier to parse logs and JavaDoc.